### ISSUE TITLE
fatal error: sweep increased allocation count

### ISSUE THREAD

**Issue by tonyghita** [neelance/graphql-go#114]
Hi, running this library on `go 1.8.3` and seeing occasional crashes with `fatal error: sweep increased allocation count`.

I _think_ this may be related to the usage of `reflect.Call()` but so far have not been able to reproduce locally. (Relevant issue: https://github.com/golang/go/issues/21717).

Has anyone else observed this error?

**Comment by tonyghita** [comment]
I found the issue. 

```graphql
type Alphabet {
 a: String!
 b: String!
 c: String!
}

type Query {
  alphabet: Alphabet!
}
```

```go
type AlphabetResolver struct {}
func (r AlphabetResolver) A() string {
  return "a"
}
func (r AlphabetResolver) B() string {
  return "b"
}
func (r AlphabetResolver) C() string {
  return "c"
}

type QueryResolver struct {}

func (r  *QueryResolver) Alphabet() AlphabetResolver {
  return AlphabetResolver{}
}
```

This causes the `reflect.Value.Call()` invocation to corrupt memory in go1.8.3 and go1.9.1.
Changing to the below fixes the zero-length return issue.

```go
func (r  *QueryResolver) Alphabet() (*AlphabetResolver, error) {
  return &AlphabetResolver{}, nil
}
```

**Comment by shurcooL** [comment]
> This causes the reflect.Value.Call() invocation to corrupt memory in go1.8.3 and go1.9.1.

Did you mean Go 1.9.0 or 1.9.1? If the latter, is this a different issue than https://github.com/golang/go/issues/21717? Is it already reported?

**Comment by tonyghita** [comment]
Ah sorry, I meant go1.9.0. The fix looks slated for the 1.9.1 milestone.

**Comment by shurcooL** [comment]
Thanks for confirming that it's not an issue in 1.9.1.
